recipes/boost-di/all/conanfile.py

```diff
@@ -0,0 +1,65 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class BoostDiConan(ConanFile):
+    name = "boost-di"
+    license = "BSL-1.0"
+    homepage = "https://github.com/boost-experimental/di"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "[Boost].DI: C++14 Dependency Injection Library."
+    topics = ("dependency-injection", "metaprogramming", "design-patterns")
+    exports_sources = ["patches/**"]
+    settings = ("compiler",)
+    options = {"with_extensions": [True, False], "diagnostics_level": [0, 1, 2]}
+    default_options = {"with_extensions": False, "diagnostics_level": 1}
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        minimal_cpp_standard = "14"
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+        minimal_version = {
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "10",
+            "Visual Studio": "15"
+        }
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
+            self.output.warn(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "di-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
```

Comment:
@uilianries, in the past you had mention patches should no be in the source method, is that still the case?

Reply:
If it's moved to another step does it means i have to remove `no_copy_source = True` as well ? Because the sources are pretty big (60 MB).

Reply:
I would also prefer them applied in the `build()` method.
Since the patch does not modify any sources, but only adds them.
Can it be applied in the `build()` method so that it creates the license file in the build folder? 
In the package folder, the copy-ing can be done from the source and build folders.

Something like:
```python
def build():
    with tools.chdir(self.build_folder):
        for patch in self.conan_data["patches"].get(self.version, []):
            tools.patch(**patch)

Reply:
This case is very peculiar, because is creating a file only. I would prefer moving it to BSL-1.0.txt file and export. Anyway, it's not possible to apply any patch on build folder, because no_copy_source avoids any copy, thus it's required to apply in source folder, or in the package folder.

Reply:
Yeah, I have seen the two solutions you are proposing. I have personally opted for applying patches in the `package`. This case with adding a license is extremely rare.
I think an export would be preferable. However this solution is acceptable :+1:

Reply:
I've changed it to use exports_sources instead.